XUtils.c

```diff
@@ -374,6 +374,20 @@ double sumPositiveValues(const double* array, size_t count) {
    return sum;
 }
 
+size_t countDigit(size_t n, size_t base) {
```

Comment:
I don't like this function. Especially when it has an int-to-float conversion that would lose precision. Can you use a simple loop instead (for decimal counting)?

```
size_t limit = 1;
unsigned int digits = 0;
for (limit = 1; limit < SIZE_MAX / 10; limit *= 10) {
   if (n < limit)
      return digits;
   digits++;
}
```

Reply:
There is a similar function for the numDigits part of other columns. This change was requested.

Reply:
@BenBE Then this should be made as a separate commit, so that I can know there are other users.
Also, why the integer-to-float cast and then cast back?

Reply:
Let me re-check with the other function. They may be subtly different so may need to be separate.

But ACK. Should be in its own commit for clarity.

Reply:
Ok, no problem. I put this function into a separate commit.